source/ide/wiki/wiki_methods.bas

```diff
@@ -6,27 +6,29 @@ FUNCTION Back2BackName$ (a$)
     Back2BackName$ = a$
 END FUNCTION
 
-FUNCTION Wiki$ (PageName$)
+FUNCTION Wiki$ (PageName$) 'Read cached wiki page (download, if not yet cached)
     Help_PageLoaded$ = PageName$
-    PageName2$ = PageName$
 
-    DO WHILE INSTR(PageName2$, " ")
-        ASC(PageName2$, INSTR(PageName2$, " ")) = 95
-    LOOP
-    DO WHILE INSTR(PageName2$, "&")
-        i = INSTR(PageName2$, "&")
-        PageName2$ = LEFT$(PageName2$, i - 1) + "%26" + RIGHT$(PageName2$, LEN(PageName2$) - i)
-    LOOP
-    DO WHILE INSTR(PageName2$, "/")
-        i = INSTR(PageName2$, "/")
-        PageName2$ = LEFT$(PageName2$, i - 1) + "%2F" + RIGHT$(PageName2$, LEN(PageName2$) - i)
-    LOOP
+    'Escape all invalid and other critical chars in filenames
+    PageName2$ = ""
+    FOR i = 1 TO LEN(PageName$)
+        c = ASC(PageName$, i)
+        SELECT CASE c
+            CASE 32 '                                    '(space)
+                PageName2$ = PageName2$ + "_"
+            CASE 34, 38, 42, 47, 58, 60, 62, 63, 92, 124 '("&*/:<>?\|)
+                PageName2$ = PageName2$ + "%" + HEX$(c)
+            CASE ELSE
+                PageName2$ = PageName2$ + CHR$(c)
+        END SELECT
+    NEXT
+    PageName3$ = wikiSafeName$(PageName2$) 'case independent name

```

Comment:
Rather than name these `PageName$`, `PageName2$` and `PageName3$`, could we go with something like  `PageNameRaw$`, `PageNameExpanded$` and `PageName$`? With the idea being that the rest of the code can just use `PageName$` and be independent of however many steps we do here. If we also put the previous logic into a function then this could just be:

`PageName$ = wikiSafeName$(urlExpandSpecialCharacters$(PageNameRaw$)))`